### PR TITLE
DAOS-4310 build: Enforce minimum pmdk build

### PR DESCRIPTION
**Pull request by brianjmurrell**
Since builds lower than 1.8-1 built with nctl which breaks DAOS.

DAOS-4352 build: Also switch from our custom fuse build to the distro fuse3 package.

Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>

### CODE REVIEW

**Comment by brianjmurrell** [comment]
Functional failure is: [DAOS-4421](https://jira.hpdd.intel.com/browse/DAOS-4421)

File: Jenkinsfile [hunk 1]
```diff
@@ -38,7 +38,7 @@
  */
 // To use a test branch (i.e. PR) until it lands to master
 // I.e. for testing library changes
-//@Library(value="pipeline-lib@your_branch") _
+@Library(value="pipeline-lib@bmurrell/dont-install-fuse") _
```
Comment (jolivier23) [inline Jenkinsfile]: I assume this is just for testing?

File: utils/rpms/daos.spec [hunk 2]
```diff
@@ -87,10 +87,14 @@ BuildRequires: libpsm_infinipath1
...
+
+%global min_pmdk 1.8-1
 %if (0%{?suse_version} >= 1500)
-Requires: libpmem1, libpmemobj1
+Requires: libpmem1 >= %{min_pmdk}, libpmemobj1 >= %{min_pmdk}, libpmemblk1 >= %{min_pmdk}
+%else
+Requires: libpmem >= %{min_pmdk}, libpmemobj >= %{min_pmdk}, libpmemblk >= %{min_pmdk}
 %endif
-Requires: fuse >= 3.4.2
+Requires: fuse3 >= 3.4.2
```
Comment (jolivier23) [inline utils/rpms/daos.spec]: The title only says pmdk

File: Jenkinsfile (hunk 1 above)
Comment (brianjmurrell) [reply Jenkinsfile]: Ah, yes.  Need to land that and remove it.  Thanks for noticing.

File: utils/rpms/daos.spec (hunk 2 above)
Comment (brianjmurrell) [reply utils/rpms/daos.spec]: Yeah, this was just an opportunistic change.

**Comment by ashleypittman** [comment]
> Should we also remove fuse from scons_local builds? @ashleypittman would that be safe?

Yes, we should probably just drop fuse as a dependency entirely, and just check that it's installed, and list in in the repo files.

**Comment by jolivier23** [comment]
Conflicting files need updating

**Comment by brianjmurrell** [comment]
Test failures are known issues: [DAOS-3945](https://jira.hpdd.intel.com/browse/DAOS-3945), [DAOS-4410](https://jira.hpdd.intel.com/browse/DAOS-4410), [DAOS-4446](https://jira.hpdd.intel.com/browse/DAOS-4446), [DAOS-4098](https://jira.hpdd.intel.com/browse/DAOS-4098)

**Comment by brianjmurrell** [comment]
> I'm not sure this is right, it seems that with this change we'd build and test against a custom version of fuse, but the rpm would install against the distro version?
> 
> I'm not against the distro version, in fact it's a task we have for 1.2 but we do need to be consistent. Perhaps I've misunderstood or am reading too much into this change?

Fixed in 0150102.

**Comment by ashleypittman** [comment]
> > I'm not sure this is right, it seems that with this change we'd build and test against a custom version of fuse, but the rpm would install against the distro version?
> > I'm not against the distro version, in fact it's a task we have for 1.2 but we do need to be consistent. Perhaps I've misunderstood or am reading too much into this change?
> 
> Fixed in [0150102](https://github.com/daos-stack/daos/commit/0150102badeb061bc21b3f15bd201581c090092a).

I was thinking more remove FUSE from build.config and utils/sl/prereq_tools/base.py entirely, and just use the distro version.  Changing the version we pull in doesn't address the issue either does it?

**Comment by brianjmurrell** [comment]
> I was thinking more remove FUSE from build.config and utils/sl/prereq_tools/base.py entirely, and just use the distro version. Changing the version we pull in doesn't address the issue either does it?

We can't do that yet.  The Ubuntu release we are using doesn't have a fuse3 yet.

**Comment by mjmac** [comment]
> 
> 
> > I was thinking more remove FUSE from build.config and utils/sl/prereq_tools/base.py entirely, and just use the distro version. Changing the version we pull in doesn't address the issue either does it?
> 
> We can't do that yet. The Ubuntu release we are using doesn't have a fuse3 yet.

Why are we allowing this ancient Ubuntu release to be a drag on progress? I don't understand the requirement to have it in the matrix. We're not testing on it, just building, right? Seems like we should disable that build until we have the resources to update to a modern release.

**Comment by brianjmurrell** [comment]
> Why are we allowing this ancient Ubuntu release to be a drag on progress?

Good question.  To the best of my understanding, we have it there mainly to satisfy the desire to be testing on a rich matrix of distros and compilers.

> I don't understand the requirement to have it in the matrix. We're not testing on it, just building, right?

Right.  On the hope/assumption that that platform (distro/compiler) will find an error that other distro/platforms are not.

> Seems like we should disable that build until we have the resources to update to a modern release.

I think the goal there is going to be to constrain ourselves to LTS releases (for everyone's sanity) and the one we are using is the most current with a new one coming (relatively) imminently.

**Comment by brianjmurrell** [comment]
Ready for [re-]review at everyone's earliest convenience.  IIRC this is a 1.0rc1 blocker.

Test failures are: [DAOS-4098](https://jira.hpdd.intel.com/browse/DAOS-4098)x2, [DAOS-4410](https://jira.hpdd.intel.com/browse/DAOS-4410), [DAOS-3945](https://jira.hpdd.intel.com/browse/DAOS-3945)

**Comment by brianjmurrell** [comment]
Third call for reviews on this 1.0 RC1 blocker, please.

**Comment by jolivier23** [comment]
Can you merge latest master?   We got green results today

File: Jenkinsfile [hunk 3]
```diff
@@ -49,7 +49,7 @@ def el7_component_repos = ""
 def component_repos = ""
 def daos_repo = "daos@${env.BRANCH_NAME}:${env.BUILD_NUMBER}"
 def el7_daos_repos = el7_component_repos + ' ' + component_repos + ' ' + daos_repo
-def functional_rpms  = "--exclude openmpi openmpi3 hwloc ndctl " +
+def functional_rpms  = "--exclude openmpi 'fuse < 3' openmpi3 hwloc ndctl " +
```
Comment (jolivier23) [inline Jenkinsfile:52]: what does 'fuse < 3' do?   should it be --exclude 'fuse < 3' ?

**Pull request merged**